src/components/download-files/DownloadFiles.js

```diff
@@ -12,40 +11,38 @@ import 'react-circular-progressbar/dist/styles.css'
 export class DownloadFiles extends React.Component {
   static propTypes = {
     files: PropTypes.object,
-    doGetDownloadLink: PropTypes.func
+    doDownloadFile: PropTypes.func
   }
 
   state = {
-    progress: null
+    isDownloading: false
   }
 
   handleOnClick = async () => {
-    const { files, doGetDownloadLink } = this.props
-    const updater = (v) => this.setState({ progress: v })
-    const { url, filename } = await doGetDownloadLink(Object.values(files))
-    downloadFile(url, filename, updater)
+    const { files, doDownloadFile } = this.props
+    this.setState({ isDownloading: true })
+
+    for (const file of Object.values(files)) {
+      const fileContent = await doDownloadFile(file.id, file.hash)
+      downloadFile(fileContent, file.name)
```

Comment:
`downloadFile` and `doDownloadFile` do very different things, but have almost the same name. Perhaps

- `doDownloadFile` -> `doGetFromIPFS`
- `downloadFile` -> `triggerSaveFile`

or anything more specific.

Reply:
Changing either one of them would be enough to make it easier to understand the difference.

Reply:
SGTM. I'll update this in #47 to avoid merge conflicts.